core/common/src/serializers/DayOfWeekSerializers.kt

```diff
@@ -16,14 +16,7 @@ import kotlinx.serialization.internal.*
  *
  * JSON example: `"MONDAY"`
  */
-@Suppress("INVISIBLE_MEMBER")
-public object DayOfWeekSerializer: KSerializer<DayOfWeek> {
-    private val impl = EnumSerializer("Month", DayOfWeek.values())
-
-    override val descriptor: SerialDescriptor
-        get() = impl.descriptor
-
-    override fun deserialize(decoder: Decoder): DayOfWeek = impl.deserialize(decoder)
-
-    override fun serialize(encoder: Encoder, value: DayOfWeek): Unit = impl.serialize(encoder, value)
-}
+public object DayOfWeekSerializer : KSerializer<DayOfWeek> by createEnumSerializer<DayOfWeek>(
+    "kotlinx.datetime.serializers.DayOfWeek",
```

Comment:
Note that it is technically a breaking change: it provide FQ name as serial name (as we do and recommend doing in the serialization) and also replaces `Month` (which is a copy-paste error I believe)

Reply:
By what principle this FQ name is built? Is it the name of the serialized type?

Reply:
According to the documentation:

>For generated serializers, serial name is equal to the corresponding class's fully-qualified name or, if overridden, SerialName. Custom serializers should provide a unique serial name that identify both the serializable class and the serializer itself, ignoring type arguments, if they are present.

We consider this one as "the default generated serializer" in fact, thus FQN of the serialized type.
I also opened https://github.com/Kotlin/kotlinx.serialization/pull/2581 to make it clear

Reply:
"FQN of the serialized type" - ok, then it should be changed as there is no type with this FQN :) 
Perhaps to "kotlinx.datetime.DayOfWeek"

Reply:
Changed it consistently in every serializer, PTAL